cpp/include/rapids_triton/triton/api/execute.hpp

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#pragma once
+#include <triton/backend/backend_common.h>
+#include <algorithm>
+#include <chrono>
+#include <cstddef>
+#include <cstdint>
+#include <rapids_triton/batch/batch.hpp>
+#include <rapids_triton/exceptions.hpp>
+#include <rapids_triton/triton/model.hpp>
+#include <rapids_triton/triton/model_instance.hpp>
+#include <rapids_triton/triton/statistics.hpp>
+#include <rapids_triton/utils/narrow.hpp>
+#include <vector>
+
+namespace triton {
+namespace backend {
+namespace rapids {
+namespace triton_api {
+template <typename ModelState, typename ModelInstanceState>
+auto* execute(TRITONBACKEND_ModelInstance* instance,
+              TRITONBACKEND_Request** raw_requests,
+              std::size_t request_count)
+{
+  auto start_time = std::chrono::steady_clock::now();
+
+  auto* result = static_cast<TRITONSERVER_Error*>(nullptr);
+
+  try {
+    auto* model_state    = get_model_state<ModelState>(*get_model_from_instance(*instance));
+    auto* instance_state = get_instance_state<ModelInstanceState>(*instance);
+    auto& model          = instance_state->get_model();
+    auto max_batch_size  = model.template get_config_param<std::size_t>("max_batch_size");
+    auto batch           = Batch(
```

Comment:
I considered that but decided not to because I thought the captures already looked somewhat suspicious (see the comment at 53) and I wanted the lambda definition location to be as clearly limited in scope as possible. But I definitely get what you mean about this being a long invocation. I'm slightly leaning toward leaving it as is, but I'm torn. What do you think?

Reply:
I'd still prefer to split this function call.